core/src/main/java/org/bitcoinj/core/Context.java

```diff
@@ -291,63 +292,68 @@ public void closeDash() {
 
         masternodePayments = null;
         masternodeSync = null;
-        initializedDash = false;
+        initializedObjects = false;
         governanceManager = null;
         masternodeListManager = null;
     }
 
-    public void initDashSync(final String directory) {
-        initDashSync(directory, null);
+    public boolean initDashSync(final String directory) {
+        return initDashSync(directory, null);
     }
 
-    public void initDashSync(final String directory, @Nullable String filePrefix) {
-        new Thread(new Runnable() {
-            @Override
-            public void run() {
-                Context.propagate(Context.this);
```

Comment:
here is another place where a thread is removed.  when initDashSync returns, loading of files is complete.